### PR TITLE
spline #266 - New rest data model - Harvester

### PR DESCRIPTION
**Pull request by lukasnalezenec**

### CODE REVIEW

File: spark/agent/src/test/scala/za/co/absa/spline/harvester/conversion/BatchReadNodeBuilderSpec.scala [hunk 1]
```diff
@@ -51,21 +52,21 @@ class BatchReadNodeBuilderSpec extends FunSpec with MockitoSugar with Matchers w
...
 
     it("should handle unrecognized source type") {
-      val bldr = new BatchReadNodeBuilder(LogicalRelation(FooBarRelation)) with FSUnawareBuilder
+      val bldr = new ReadNodeBuilder(LogicalRelation(FooBarRelation)) with FSUnawareBuilder
 
       val readOp = bldr.build()
 
-      readOp.sourceType shouldEqual "???: za.co.absa.spline.harvester.conversion.FooBarRelation$"
-      readOp.sources shouldBe empty
+      readOp.params.getOrElse(Constants.SOURCE_TYPE, fail) shouldEqual "???: za.co.absa.spline.harvester.conversion.FooBarRelation$"
```
Comment (leborgneromain) [inline spark/agent/src/test/scala/za/co/absa/spline/harvester/conversion/BatchReadNodeBuilderSpec.scala]: why is there "???" in the source type ?

File: spark/agent/src/main/scala/za/co/absa/spline/harvester/componentCreators.scala [hunk 2]
```diff
@@ -146,64 +149,56 @@ object ExpressionConverter {
...
-        e.prettyName,
-        getDataType(e).id,
-        e.children map convert,
-        getExpressionSimpleClassName(e),
-        getExpressionExtraParameters(e))
+  override type To = Expression
+
+  override def convert(sparkExpr: SparkExpression): Expression = sparkExpr match {
+
+//    case a: expressions.Alias =>
```
Comment (leborgneromain) [inline spark/agent/src/main/scala/za/co/absa/spline/harvester/componentCreators.scala]: Why Keep all this commented code ?

File: spark/agent/src/main/scala/za/co/absa/spline/harvester/componentCreators.scala [hunk 3]
```diff
@@ -63,16 +65,17 @@ class DataTypeConverter extends AbstractConverter {
...
-  override type From = SparkAttribute
-  override type To = Attribute
-
-  override def convert(attr: SparkAttribute): Attribute = {
-    Attribute(
-      id = randomUUID,
-      name = attr.name,
-      dataTypeId = dataTypeConverter.convert(attr.dataType, attr.nullable).id)
-  }
+//  extends AbstractConverter
```
Comment (leborgneromain) [inline spark/agent/src/main/scala/za/co/absa/spline/harvester/componentCreators.scala]: Why Keep all this commented code ?

File: spark/spark-adapter-2.4/src/main/scala/za/co/absa/spline/sparkadapterapi/WriteCommandParserImpl.scala [hunk 4]
```diff
@@ -17,65 +17,112 @@
...
+            val clusterUrl = sparkContext.getConf.getOption("spark.master")
 
-  override def asWriteCommand(operation: LogicalPlan): AbstractWriteCommand = {
-    val op = operation.asInstanceOf[CreateDataSourceTableAsSelectCommand]
+            val identifier = op.table.storage.locationUri match {
+              case Some(location) => location.toURL.toString
+              case _ =>
+                val codec = new URLCodec()
+                URIPrefixes.managedTablePrefix +
+                  codec.encode(clusterUrl.getOrElse(throw new IllegalArgumentException("Unknown cluster name."))) + ":" +
```
Comment (leborgneromain) [inline spark/spark-adapter-2.4/src/main/scala/za/co/absa/spline/sparkadapterapi/WriteCommandParserImpl.scala]: String Interpolation

File: spark/agent/src/main/scala/za/co/absa/spline/harvester/componentCreators.scala [hunk 5]
```diff
@@ -146,64 +149,56 @@ object ExpressionConverter {
   }
 }
 
+
+
 class ExpressionConverter(dataTypeConverter: DataTypeConverter, attributeConverter: AttributeConverter)
-  extends AbstractConverter {
+  extends AbstractConverter with ExpressionConverterIface{
```
Comment (wajda) [inline spark/agent/src/main/scala/za/co/absa/spline/harvester/componentCreators.scala]: Why using `Iface` suffix?

File: spark/agent/src/main/scala/za/co/absa/spline/harvester/listener/StructuredStreamingListener.scala [hunk 6]
```diff
@@ -93,17 +93,17 @@ class StructuredStreamingListener(
   private def processExecution(se: StreamExecution): Unit = {
     assume(se.logicalPlan.resolved, "we harvest lineage from analyzed logic plans")
     val logicalPlan = ReflectionUtils.getFieldValue[LogicalPlan](se, "analyzedPlan")
-    val logicalPlanLineage = lineageBuilderFactory.createBuilder(logicalPlan, None, se.sparkSession.sparkContext).buildLineage()
+    val executionPlan = lineageBuilderFactory.createBuilder(logicalPlan, None, se.sparkSession.sparkContext).buildExecutionPlan
```
Comment (wajda) [inline spark/agent/src/main/scala/za/co/absa/spline/harvester/listener/StructuredStreamingListener.scala]: Empty-paren method accessed as parameterless

**Pull request closed without merging**